test/unit/lib/classes/Variables.test.js

```diff
@@ -2780,4 +2708,113 @@ module.exports = {
       expect(processedConfig.custom.nestedReference).to.equal('resolvedNested');
     });
   });
+
+  describe('variable resolving', () => {
+    describe('when unresolvedVariablesNotificationMode is set to "error"', () => {
+      it('should error for missing "environment variable" type variables', async () => {
+        await expect(
+          runServerless({
+            fixture: 'variables',
+            cliArgs: ['print'],
+            configExt: {
+              unresolvedVariablesNotificationMode: 'error',
+              custom: { myVariable: '${env:missingEnvVar}' },
+            },
+          })
+        ).to.be.rejectedWith('environment variable');
```

Comment:
Let's rely on error codes as e.g. here: https://github.com/serverless/serverless/blob/4d484684c15171af3dad3b1e26c1d3248c0b90ae/test/unit/lib/plugins/aws/package/compile/events/httpApi.test.js#L456-L459

And we can use same error code for all